backend/booking/serializers.py

```diff
@@ -191,41 +192,108 @@ def get_reviews(self, obj):
         return serializer.data
 
     def get_next_availabilty(self, obj):
-        now = datetime.datetime.now()
-
-        next_slot = Slot.objects.filter(
-            days=now.strftime('%A'),
-            start_time__gte=now.time(),
-            is_booked=False,
-            location=obj
-        ).order_by('start_time').first()
-
-        next_additional_slot = AdditionalSlot.objects.filter(
-            date__gte=now.date(),
-            is_active=True,
-            location=obj
-        ).order_by('date', 'start_time').first()
-
-        next_unavailable_slot = UnavailableSlot.objects.filter(
-            date__gte=now.date(),
-            is_active=True,
-            location=obj
-        ).order_by('date', 'start_time').first()
-
-        if next_slot and next_additional_slot:
-            if next_slot.start_time < next_additional_slot.start_time:
-                if next_slot.start_time < next_unavailable_slot.start_time:
-                    return SlotSerializer(next_slot).data
-                else:
-                    return SlotSerializer(next_slot).data
-            else:
-                return AdditionalSlotSerializer(next_additional_slot).data
-        elif next_slot:
-            return SlotSerializer(next_slot).data
-        elif next_additional_slot:
-            return AdditionalSlotSerializer(next_additional_slot).data
+        now = datetime.datetime.now().replace(microsecond=0)
+        current_time = (now + datetime.timedelta(hours=1)).time()
+        selected_date = now.date()
 
-        return None
+        # Find the first court for the given OrganizationLocation
+        try:
+            court = Court.objects.filter(location=obj).first()
+            if not court:
+                return None
+        except Court.DoesNotExist:
+            return None
+
+        # Exclude unavailable and booked slots
+        unavailable_slots = UnavailableSlot.objects.filter(
+            court=court,
+            date__gte=selected_date,
+            is_active=True
+        ).values_list('start_time', 'end_time')
+        
+        booked_slots = Booking.objects.filter(
+            court=court,
+            booking_date__gte=selected_date
+        ).values_list('slot__start_time', 'slot__end_time')
+        
+        excluded_times = list(unavailable_slots) + list(booked_slots)
+
+        nearest_slots = []
+
+        # Find nearest slot from Slot table
+        for i in range(0, 7):  # Search for up to one week
+            target_date = now + datetime.timedelta(days=i)
+            target_weekday = target_date.strftime('%A')
+
+            slots = Slot.objects.filter(
+                court=court,
+                days=target_weekday,
+                start_time__gte=(current_time if i == 0 and selected_date == now.date() else datetime.datetime.min.time())
```

Comment:
Don't we have is_active flag here for Slot? If yes, then both Slot & AdditionalSlot forloop can be combined in a single function (may be util) and use that here!

Reply:
slot follow days and additional slot follow date, slot has is_booked flag yes, Ok will create and use util function.